docs/technical_reference/unit_models/treatment_models/chem_soft.rst

```diff
@@ -0,0 +1,255 @@
+
+   "Number of mixers", "``no_of_mixer``", ":math:`n_{mixer}`", "", ":math:`\text{dimensionless}`"
+   "Number of flocculators", "``no_of_floc``", ":math:`n_{floc}`", "", ":math:`\text{dimensionless}`"
+   "Retention time of mixer", "``retention_time_mixer``", ":math:`RT_{mixer}`", "0.1-5", ":math:`\text{min}`"
+   "Retention time of flocculator", "``retention_time_floc``", ":math:`RT_{floc}`", "10-45", ":math:`\text{min}`"
+   "Retention time of sedimentation basin", "``retention_time_sed``", ":math:`RT_{sed}`", "120-240",  ":math:`\text{min}`"
+   "Retention time of recarbonation basin", "``retention_time_recarb``", ":math:`RT_{recarb}`", "15-30", ":math:`\text{min}`"
+   "Fractional volume recovery", "``frac_vol_recovery``", "", "", ":math:`\text{dimensionless}`"
+   "Removal efficiency of components (except Ca2+ and Mg2+)", "``removal_efficiency``", "","",":math:`\text{dimensionless}`"
+   "CO2 dose in CaCO3 equivalents", "``CO2_CaCO3``",":math:`CO_{2,CaCO_{3}-hardness}`","", ":math:`\text{g/}\text{L}`"
+   "Velocity gradient in mixer", "``vel_gradient_mix``", ":math:`\text{velocity gradient}_{mixer}`", "300-1000",":math:`\text{/}\text{s}`"
+   "Velocity gradient in flocculator", "``vel_gradient_floc``", ":math:`\text{velocity gradient}_{floc}`", "20-80", ":math:`\text{/}\text{s}`"
+
+The following variables should be fixed to 0 if their dose is not calculated in the softening procedure for the model to be fully specified. 
+The softening procedure where the doses are calculated in are listed in the table.
+
+.. csv-table::
+   :header: "Variables", "Softening procedure", "Variable Name", "Symbol", "Unit"
+
+   "Excess lime", "excess_lime, excess_lime_soda", "``excess_CaO``", ":math:`CaO`", ":math:`\text{g/}\text{L}`"
+   "Soda ash","single_stage_lime_soda, excess_lime_soda ", "``Na2CO3_dosing``", ":math:`Na_{2}CO_{3}`", ":math:`\text{g/}\text{L}`" 
+   "CO2 dose in second basin","excess_lime_soda", "``CO2_second_basin``", ":math:`CO_{2,second-basin}`", ":math:`\text{g/}\text{L}`" 
+   "MgCl2","Silica removal", "``MgCl2_dosing``", ":math:`MgCl_{2}`", ":math:`\text{g/}\text{L}`" 
+
+
+Solution Composition
+---------------
+
+This chemical softening model requires an input solute list from the user. Components that must be included
+are shown in the code below. Additional elements may added.
+
+.. code-block::
+   
+   component_list = ["Ca_2+","Mg_2+","Alkalinity_2-"]
+
+A default removal efficiency is assumed for components (other than :math:`\text{Ca}^{2+}` and :math:`\text{Mg}^{2+}`).
+Users can provide an input dictionary of removal efficiencies for each of the components.
+
+.. code-block::
+
+   removal_eff_dict = dict(
+            zip([
+                x for x in component_list if x not in ["Ca_2+","Mg_2+"]
+                ]
+                ,
+                [   
+                    0.7 if j != "TDS" else 1e-3
+                    for j in component_list 
```

Comment:
Why is TDS specified here? It could technically be anything, right? If this is meant as an arbitrary example, you should probably still include TDS in the component list above.